celery/backends/redis.py

```diff
@@ -413,18 +413,21 @@ def apply_chord(self, header_result, body, **kwargs):
     def on_chord_part_return(self, request, state, result,
                              propagate=None, **kwargs):
         app = self.app
-        tid, gid = request.id, request.group
+        tid, gid, group_index = request.id, request.group, request.group_index
```

Comment:
There is some breakage when we run a worker on this branch but issue tasks from an app (using the example app code from #4858) using celery master:

```
[2020-07-13 09:14:51,351: ERROR/MainProcess] Pool callback raised exception: AttributeError("'Context' object has no attribute 'group_index'")
Traceback (most recent call last):
  File "/dev/celery/venv/lib64/python3.8/site-packages/billiard/pool.py", line 1796, in safe_apply_callback
    fun(*args, **kwargs)
  File "/dev/celery/celery/worker/request.py", line 531, in on_failure
    self.task.backend.mark_as_failure(
  File "/dev/celery/celery/backends/base.py", line 172, in mark_as_failure
    self.on_chord_part_return(request, state, exc)
  File "/dev/celery/celery/backends/redis.py", line 416, in on_chord_part_return
    tid, gid = request.id, request.group
AttributeError: 'Context' object has no attribute 'group_index'
```

From what I can tell, when running a worker node on master and a task issuer on this branch, the behaviour simply falls back to having the results be randomly ordered based on order of completion. That seems like a fine graceful degradation.

Reply:
This is fixed by adding the `Context.group_index` attribute and handling passing of `group_index` when uplifting to chords. That should also fix the missing coverage on the `Request.group_index` property method.